src/kubernetes.cc

```diff
@@ -961,6 +961,38 @@ json::value KubernetesReader::FindTopLevelOwner(
   return FindTopLevelOwner(ns, GetOwner(ns, ref->As<json::Object>()));
 }
 
+bool KubernetesReader::ValidatePodConnectivity() const {
+  try {
+    QueryMaster(std::string(kKubernetesEndpointPath) + "/pods?limit=1");
+
+    return true;
+  } catch (const QueryException& e) {
+    // Already logged.
+    return false;
+  }
+}
+
+bool KubernetesReader::ValidateNodeConnectivity() const {
+  const std::string node_name = CurrentNode();
+  if (node_name.empty()) {
+    return false;
+  }
+
+  try {
+    json::value raw_node = QueryMaster(
```

Comment:
Didn't we talk about doing this test first? You don't need a node name, just limit it to one (random) node.
Also, the `raw_node` value is unused, so just ignore the return value of `QueryMaster` instead.

Reply:
Done.